File: app/src/main/java/com/crossbowffs/quotelock/app/CommonReceiver.java
package com.crossbowffs.quotelock.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.net.ConnectivityManager;

import com.crossbowffs.quotelock.consts.PrefKeys;
import com.crossbowffs.quotelock.utils.WorkUtils;
import com.crossbowffs.quotelock.utils.Xlog;

public class CommonReceiver extends BroadcastReceiver {
    private static final String TAG = CommonReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent.getAction();
        Xlog.d(TAG, "Received action: %s", action);
        if (Intent.ACTION_BOOT_COMPLETED.equals(action) || ConnectivityManager.CONNECTIVITY_ACTION.equals(action)) {
            if (Intent.ACTION_BOOT_COMPLETED.equals(action)) {
                // Notify LockscreenHook to show current quotes after booting.
                notifyBooted(context);
            }
            WorkUtils.createQuoteDownloadWork(context, false);
        }
    }

    private void notifyBooted(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PrefKeys.PREF_QUOTES, Context.MODE_PRIVATE);
        boolean hasBootNotifyFlag = sharedPreferences.contains(PrefKeys.PREF_BOOT_NOTIFY_FLAG);
        if (!hasBootNotifyFlag) {
            sharedPreferences.edit().putInt(PrefKeys.PREF_BOOT_NOTIFY_FLAG, 0).apply();
        } else {
            sharedPreferences.edit().remove(PrefKeys.PREF_BOOT_NOTIFY_FLAG).apply();
        }
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/app/CommonReceiver.kt
package com.crossbowffs.quotelock.app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.crossbowffs.quotelock.data.ConfigurationRepository
import com.crossbowffs.quotelock.data.modules.QuoteRepository
import com.crossbowffs.quotelock.utils.WorkUtils
import com.crossbowffs.quotelock.utils.Xlog
import com.crossbowffs.quotelock.utils.className
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject

class CommonReceiver : BroadcastReceiver(), KoinComponent {

    private val quoteRepository: QuoteRepository by inject()

    private val configurationRepository: ConfigurationRepository by inject()

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        Xlog.d(TAG, "Received action: %s", action ?: "-")
        if (Intent.ACTION_BOOT_COMPLETED == action) {
            // Notify LockscreenHook to show current quotes after booting.
            quoteRepository.notifyBooted()
            WorkUtils.createQuoteDownloadWork(
                context,
                configurationRepository.refreshInterval,
                configurationRepository.isRequireInternet,
                configurationRepository.isUnmeteredNetworkOnly,
                false
            )
        }
    }

    companion object {
        private val TAG = className<CommonReceiver>()
    }
}